### PR TITLE
Review suggesions

### PR DESCRIPTION
**Pull request by dkjellin**
I've made some changes, I think this is an improvement on your PR. I did not write any additional tests so if you accept this (or parts of this) do make a decision on how much additional testing is needed.

### CODE REVIEW

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    public void postBuildStatus(AbstractBuild build, TaskListener listener) {
+        BitbucketRevisionAction revisionAction = build.getAction(BitbucketRevisionAction.class);
+        if (revisionAction == null) {
+            return;
+        }
+        Optional<BitbucketServerConfiguration> serverOptional = pluginConfiguration.getServerById(revisionAction.getServerId());
+        if (serverOptional.isPresent()) {
+            BitbucketServerConfiguration server = serverOptional.get();
+            listener.getLogger().println("Posting build status to: " + server.getServerName());
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: we may want to include the status here? I don't know. I went for basics first. But happy to have something richer.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketRevisionAction.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import hudson.model.Action;
+
+import javax.annotation.CheckForNull;
+
+public class BitbucketRevisionAction implements Action {
+
+    private final String revisionSha1;
+    private final String serverId;
+
+    public BitbucketRevisionAction(String revisionSha1, String serverId) {
```
Comment (mhenschke-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketRevisionAction.java]: This seems like a really good solution to me- it means we can remove `getLatestRevision` from the SCM entirely.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java (hunk 1 above)
Comment (mhenschke-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: I'm flexible. Given the feature is inconsequential to the build itself I'm actually a bit hesitant to include too much detail in the logs, but the enum status seems like a reasonable compromise?

Comment (mhenschke-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: Something like:
```suggestion
            BitbucketBuildStatus buildStatus = new BitbucketBuildStatus(build)
            listener.getLogger().println("Posting build status of " + buildStatus.getState() " to: " + server.getServerName());
```

Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: yeah, I'm happy either way. I just want something in the _build_ logs to show that we attempted to post the build status. The enum seems easy enough, and we have to get hold of it anyway.

Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: yep (and then not create it further down), happy for you to merge this PR and make needful changes as you see fit.

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+            BitbucketCredentials credentials =
+                    BitbucketCredentialsAdaptor.createWithFallback(server.getCredentials(), server);
+            try {
+                bitbucketClientFactoryProvider.getClient(server.getBaseUrl(), credentials)
+                        .getBuildStatusClient(revisionAction.getRevisionSha1())
+                        .post(new BitbucketBuildStatus(build));
+                return;
+            } catch (BitbucketClientException e) {
```
Comment (mhenschke-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPoster.java]: Widened this slightly to `RuntimeException`

**Pull request merged**